.gitignore

```diff
@@ -28,3 +28,7 @@ test/iterate
 test/case
 test/custom
 /tmp/
+mingw_static
+mingw_shared
+msvc_shared
+msvc_static
```

Comment:
Probably these should not be here?

Reply:
When compiling with AppVeyor see appveyor.yml, these directories are created, but you don't want to see it in git, because you you will not commit the files in these folders.

Reply:
I'm not sure we care about what git sees on AppVeyor, but fine.

Reply:
(Similarly, we don't add `build_shared` even though we use this directory on Travis.)